aten/src/ATen/native/TensorIndexing.cpp

```diff
@@ -154,5 +221,208 @@ std::ostream& operator<<(std::ostream& stream, const std::vector<TensorIndex>& t
   return stream;
 }
 
+// This mirrors `count_specified_dimensions` in torch/csrc/autograd/python_variable_indexing.cpp
```

Comment:
Mirrors, or replaces? :)

Reply:
Nooooooooooooooooo XD

Reply:
Can you explain why these can't be deduplicated?

Reply:
(Naively, I would have expected that once you push everything down, the Python bindings shouldn't need to know about this anymore.)

Reply:
The main reason is `count_specified_dimensions` is on the hot path of indexing (i.e. it's called by `applySlicing` which is called by `THPVariable_getitem` when handling indexing of more than one dimension). If we were to merge the Python/C++ `count_specified_dimensions` function, on the Python side we would have to construct a `std::vector` container to be consumed by the C++ `count_specified_dimensions` function, which adds 100s of nanoseconds overhead and is undesirable.

Reply:
OK, that should be written in the code comments